sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1793,6 +1794,7 @@ class Analyzer(
    * Replaces [[UnresolvedFunction]]s with concrete [[Expression]]s.
    */
   object ResolveFunctions extends Rule[LogicalPlan] {
+    val trimWarningEnabled = new AtomicBoolean(true)
```

Comment:
When do you set it back to `true`? The `ResolveFunctions` object can live forever. By setting it to `false` only once, you disable warnings for all apps and users.

Let's image multiple notebooks connected to the same cluster - they will share the same `trimWarningEnabled` if I am not wrong.

Reply:
This is an inner object of `Analyzer`, and we will have a new instance for a new session which creates an `Analyzer`.